services/blockstorage/test/commit_block_test.go

```diff
@@ -14,6 +15,8 @@ import (
 func TestCommitBlockSavesToPersistentStorage(t *testing.T) {
 	test.WithContext(func(ctx context.Context) {
 		harness := newHarness(ctx)
+		// adding the broadcast as it might hit because of timeout, its not required for the test specifically
```

Comment:
WHAAAAAT?
why? 
please explain

Reply:
i think this is something we can discuss at the weekly:
originally the harness of block storage (which was written 3 months ago?) had a lot of bad assumptions, one of which was just having that global broadcast mock - while most tests do need it, until the sync process no-one really cared about it.
so the sync is part of the 'global' harness and then if a test is taking too long to trigger a sync process, it will begin calling the broadcast, so unless the test is specifically not expecting the sync to start (because of no commits), it probably needs this broadcast mock or it will flake out - this is true now of any test that is starting block storage (and not a mock, but a mock gossip)

Reply:
so maybe you don't start block sync in component tests of the block storage?

Reply:
putting aside the technical 'how-to' on doing that, i think the sync should be running in the background of the component tests, as it will in production - the alternative is to redesign the sync as its own service

Reply:
change the harness to configure sync to a very long timeout